src/main/java/roboy/dialog/states/ordinaryStates/FUAState.java

```diff
@@ -0,0 +1,103 @@
+package roboy.dialog.states.ordinaryStates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import roboy.context.Context;
+import roboy.context.contextObjects.IntentValue;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.memory.Neo4jRelationships;
+import roboy.memory.nodes.Interlocutor;
+import roboy.memory.nodes.MemoryNodeModel;
+import roboy.util.QAJsonParser;
+import roboy.util.RandomList;
+
+import java.util.List;
+import java.util.Set;
+
+import static roboy.memory.Neo4jRelationships.*;
+
+/**
+ * Follow Up Asking State
+ * This state is only entered if there are some known facts about the active interlocutor.
+ * It asks if the known facts are still up to date.
+ */
+public class FUAState extends State {
+
+    private QAJsonParser qaValues;
+    private Neo4jRelationships[] predicates = { FROM, HAS_HOBBY, WORK_FOR, STUDY_AT };
+    private Neo4jRelationships selectedPredicate;
+    private State nextState;
+
+    private final String TRANSITION_INFO_UPDATED = "questionAnswering";
+    private final Logger LOGGER = LogManager.getLogger();
+
+    public FUAState(String stateIdentifier, StateParameters params) {
+        super(stateIdentifier, params);
+        qaValues = new QAJsonParser(params.getParameter("qaFile"));
+    }
+
+    @Override
+    public State.Output act() {
+        Interlocutor person = Context.getInstance().ACTIVE_INTERLOCUTOR.getValue();
+
+        for (Neo4jRelationships predicate : predicates) {
+            if (person.hasRelationship(predicate) && !Context.getInstance().DIALOG_INTENTS.contains(new IntentValue("FUP", predicate))) {
+                selectedPredicate = predicate;
+                break;
+            }
+        }
+
+        if (selectedPredicate != null) {
+            RandomList<String> questions = new RandomList<>(qaValues.getFollowUpQuestions(selectedPredicate));
+            String retrievedResult = "";
+            RandomList<MemoryNodeModel> nodes = retrieveNodesFromMemoryByIds(person.getRelationships(selectedPredicate));
+            if (!nodes.isEmpty()) {
+                retrievedResult = nodes.getRandomElement().getProperties().get("name").toString();
+            }
+            String question = String.format(questions.getRandomElement(), retrievedResult);
+            Context.getInstance().DIALOG_INTENTS_UPDATER.updateValue(new IntentValue("FUP", selectedPredicate));
```

Comment:
``new IntentValue("FUP"`` 
"FUP" is used as a state identifier here (first argument of ``IntentValue`` is a state identifier). Actually, state identifiers are set in the personality file and can be very different from "FUP". You can get the real identifier by ``getIdentifier()``. 

If you use "FUP" intentionally instead of the real identfier for some reason, please add a comment.

Reply:
Refactored

Reply:
It is intentional